### DISCUSSION TITLE
Error

### DISCUSSION THREAD

**Discussion by giox851** [saleweaver/python-amazon-sp-api#1610] [FAQ] [answered]
I'm doing a test

from datetime import datetime, timedelta
from sp_api.base import Marketplaces
from sp_api.api import Orders
from sp_api.util import throttle_retry, load_all_pages

credentials = dict(
    refresh_token='xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxMfMEJLOm4kdb8rP3z77VHlgMq8',
    lwa_app_id='xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx4d54b590c222b73ec3b4',
    lwa_client_secret='xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxd8d42dbdd046657497d13'
)
SP_API_DEFAULT_MARKETPLACE="IT"
AWS_ENV="SANDBOX"
ENV_DISABLE_DONATION_MSG=1 

Orders(credentials=credentials).get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=1)).isoformat())

I get the following error:

 File "c:\Temp\test_amazon.py", line 15, in <module>
    Orders(credentials=credentials).get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=1)).isoformat())
  File "C:\Temp\Whist\myenv\lib\site-packages\sp_api\base\helpers.py", line 21, in wrapper
    return function(*args, **kwargs)
  File "C:\Temp\Whist\myenv\lib\site-packages\sp_api\api\orders\orders.py", line 58, in get_orders
    return self._request(kwargs.pop('path'), params={**kwargs})
  File "C:\Temp\Whist\myenv\lib\site-packages\sp_api\base\client.py", line 119, in _request
    return self._check_response(res, res_no_data, bulk, wrap_list)
  File "C:\Temp\Whist\myenv\lib\site-packages\sp_api\base\client.py", line 145, in _check_response
    raise exception(error, headers=res.headers)
sp_api.base.exceptions.SellingApiForbiddenException: [{'code': 'Unauthorized', 'message': 'Access to requested resource is denied.', 'details': ''}]

what am I doing wrong?

thanks

**Comment by saleweaver** [comment, marked as answer]
You should remove this line:

ENV_DISABLE_DONATION_MSG=1

And donate instead. 

Jokes aside, the sandbox allows for a certain request object, and isn't suitable for real world testing.

**Reply by giox851** [reply]
even indicating the official market I receive the same error!!
